safe-discriminant-derive/src/lib.rs

```diff
@@ -107,3 +124,26 @@ pub fn derive_discriminant(item: TokenStream) -> TokenStream {
         Ok(s) => s,
     }
 }
+
+#[cfg(feature = "test-utils")]
```

Comment:
Optional/Nit: Slight preference to put this in a separate module and just `#[cfg(feature = "test-utils")]` the module instead of doing it per testonly attribute.

I think it'd read a lot clearer in tests too:

```rs
use safe_discriminant::Discriminant;
use safe_discriminant_derive::testonly::repr;

#[derive(Discriminant)]
#[repr(u8)]
pub enum Foo {
    A = 0,
    B = 1,
}

fn main() {}
```